### ISSUE TITLE
Warnings on Frontier

### ISSUE THREAD

**Issue by bishtgautam** [RDycore/RDycore#187]
These are warnings on Frontier, but I also see similar warnings on Perlmutter.

```
/lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/src/rdydata.c:68:56: warning: argument 5 of type 'PetscReal[(PetscInt)num_edges * (PetscInt)ndof]' {aka 'double[(PetscInt)num_edges * (PetscInt)ndof]'} declared with mismatched bound '(PetscInt) num_edges * (PetscInt) ndof' [-Wvla-parameter]
   68 |                                              PetscReal values[num_edges * ndof]) {
      |                                              ~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~
In file included from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/include/private/rdymmsconfigimpl.h:6,
                 from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/include/private/rdyconfigimpl.h:8,
                 from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/include/private/rdycoreimpl.h:6,
                 from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/src/rdydata.c:1:
/lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/build-frontier-gpu-opt-32bit-gcc-11-2-0-fc288817/include/rdycore.h:99:156: note: previously declared as 'PetscReal[(PetscInt)ndof * (PetscInt)num_edges]' {aka 'double[(PetscInt)ndof * (PetscInt)num_edges]'} with bound '(PetscInt) ndof * (PetscInt) num_edges'
   99 | PETSC_EXTERN PetscErrorCode RDySetDirichletBoundaryValues(RDy rdy, const PetscInt boundary_index, const PetscInt num_edges, const PetscInt ndof, PetscReal values[ndof * num_edges]);
      |                                                                                                                                                  ~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~
/lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/src/rdydata.c:131:79: warning: argument 3 of type 'PetscReal[size]' {aka 'double[size]'} declared with mismatched bound 'size' [-Wvla-parameter]
  131 | PetscErrorCode RDyGetLocalCellHeights(RDy rdy, const PetscInt size, PetscReal values[size]) {
      |                                                                     ~~~~~~~~~~^~~~~~~~~~~~
In file included from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/include/private/rdymmsconfigimpl.h:6,
                 from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/include/private/rdyconfigimpl.h:8,
                 from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/include/private/rdycoreimpl.h:6,
                 from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/src/rdydata.c:1:
/lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/build-frontier-gpu-opt-32bit-gcc-11-2-0-fc288817/include/rdycore.h:79:92: note: previously declared as 'PetscReal[size]' {aka 'double[size]'} with bound 'size'
   79 | PETSC_EXTERN PetscErrorCode RDyGetLocalCellHeights(RDy rdy, const PetscInt size, PetscReal values[size]);
      |                                                                                  ~~~~~~~~~~^~~~~~~~~~~~
/lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/src/rdydata.c:139:82: warning: argument 3 of type 'PetscReal[size]' {aka 'double[size]'} declared with mismatched bound 'size' [-Wvla-parameter]
  139 | PetscErrorCode RDyGetLocalCellXMomentums(RDy rdy, const PetscInt size, PetscReal values[size]) {
      |                                                                        ~~~~~~~~~~^~~~~~~~~~~~
In file included from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/include/private/rdymmsconfigimpl.h:6,
                 from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/include/private/rdyconfigimpl.h:8,
                 from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/include/private/rdycoreimpl.h:6,
                 from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/src/rdydata.c:1:
/lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/build-frontier-gpu-opt-32bit-gcc-11-2-0-fc288817/include/rdycore.h:80:95: note: previously declared as 'PetscReal[size]' {aka 'double[size]'} with bound 'size'
   80 | PETSC_EXTERN PetscErrorCode RDyGetLocalCellXMomentums(RDy rdy, const PetscInt size, PetscReal values[size]);
      |                                                                                     ~~~~~~~~~~^~~~~~~~~~~~
/lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/src/rdydata.c:147:82: warning: argument 3 of type 'PetscReal[size]' {aka 'double[size]'} declared with mismatched bound 'size' [-Wvla-parameter]
  147 | PetscErrorCode RDyGetLocalCellYMomentums(RDy rdy, const PetscInt size, PetscReal values[size]) {
      |                                                                        ~~~~~~~~~~^~~~~~~~~~~~
In file included from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/include/private/rdymmsconfigimpl.h:6,
                 from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/include/private/rdyconfigimpl.h:8,
                 from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/include/private/rdycoreimpl.h:6,
                 from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/src/rdydata.c:1:
/lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/build-frontier-gpu-opt-32bit-gcc-11-2-0-fc288817/include/rdycore.h:81:95: note: previously declared as 'PetscReal[size]' {aka 'double[size]'} with bound 'size'
   81 | PETSC_EXTERN PetscErrorCode RDyGetLocalCellYMomentums(RDy rdy, const PetscInt size, PetscReal values[size]);
      |                                                                                     ~~~~~~~~~~^~~~~~~~~~~~
/lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/src/rdydata.c:174:87: warning: argument 3 of type 'PetscReal[size]' {aka 'double[size]'} declared with mismatched bound 'size' [-Wvla-parameter]
  174 | PetscErrorCode RDySetWaterSourceForLocalCells(RDy rdy, const PetscInt size, PetscReal values[size]) {
      |                                                                             ~~~~~~~~~~^~~~~~~~~~~~
In file included from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/include/private/rdymmsconfigimpl.h:6,
                 from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/include/private/rdyconfigimpl.h:8,
                 from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/include/private/rdycoreimpl.h:6,
                 from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/src/rdydata.c:1:
/lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/build-frontier-gpu-opt-32bit-gcc-11-2-0-fc288817/include/rdycore.h:100:100: note: previously declared as 'PetscReal[size]' {aka 'double[size]'} with bound 'size'
  100 | PETSC_EXTERN PetscErrorCode RDySetWaterSourceForLocalCells(RDy rdy, const PetscInt size, PetscReal values[size]);
      |                                                                                          ~~~~~~~~~~^~~~~~~~~~~~
/lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/src/rdydata.c:189:91: warning: argument 3 of type 'PetscReal[size]' {aka 'double[size]'} declared with mismatched bound 'size' [-Wvla-parameter]
  189 | PetscErrorCode RDySetXMomentumSourceForLocalCells(RDy rdy, const PetscInt size, PetscReal values[size]) {
      |                                                                                 ~~~~~~~~~~^~~~~~~~~~~~
In file included from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/include/private/rdymmsconfigimpl.h:6,
                 from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/include/private/rdyconfigimpl.h:8,
                 from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/include/private/rdycoreimpl.h:6,
                 from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/src/rdydata.c:1:
/lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/build-frontier-gpu-opt-32bit-gcc-11-2-0-fc288817/include/rdycore.h:101:104: note: previously declared as 'PetscReal[size]' {aka 'double[size]'} with bound 'size'
  101 | PETSC_EXTERN PetscErrorCode RDySetXMomentumSourceForLocalCells(RDy rdy, const PetscInt size, PetscReal values[size]);
      |                                                                                              ~~~~~~~~~~^~~~~~~~~~~~
/lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/src/rdydata.c:203:91: warning: argument 3 of type 'PetscReal[size]' {aka 'double[size]'} declared with mismatched bound 'size' [-Wvla-parameter]
  203 | PetscErrorCode RDySetYMomentumSourceForLocalCells(RDy rdy, const PetscInt size, PetscReal values[size]) {
      |                                                                                 ~~~~~~~~~~^~~~~~~~~~~~
In file included from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/include/private/rdymmsconfigimpl.h:6,
                 from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/include/private/rdyconfigimpl.h:8,
                 from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/include/private/rdycoreimpl.h:6,
                 from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/src/rdydata.c:1:
/lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/build-frontier-gpu-opt-32bit-gcc-11-2-0-fc288817/include/rdycore.h:102:104: note: previously declared as 'PetscReal[size]' {aka 'double[size]'} with bound 'size'
  102 | PETSC_EXTERN PetscErrorCode RDySetYMomentumSourceForLocalCells(RDy rdy, const PetscInt size, PetscReal values[size]);
      |                                                                                              ~~~~~~~~~~^~~~~~~~~~~~
/lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/src/rdydata.c:233:82: warning: argument 3 of type 'PetscReal[size]' {aka 'double[size]'} declared with mismatched bound 'size' [-Wvla-parameter]
  233 | PetscErrorCode RDyGetLocalCellXCentroids(RDy rdy, const PetscInt size, PetscReal values[size]) {
      |                                                                        ~~~~~~~~~~^~~~~~~~~~~~
In file included from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/include/private/rdymmsconfigimpl.h:6,
                 from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/include/private/rdyconfigimpl.h:8,
                 from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/include/private/rdycoreimpl.h:6,
                 from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/src/rdydata.c:1:
/lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/build-frontier-gpu-opt-32bit-gcc-11-2-0-fc288817/include/rdycore.h:83:95: note: previously declared as 'PetscReal[size]' {aka 'double[size]'} with bound 'size'
   83 | PETSC_EXTERN PetscErrorCode RDyGetLocalCellXCentroids(RDy rdy, const PetscInt size, PetscReal values[size]);
      |                                                                                     ~~~~~~~~~~^~~~~~~~~~~~
/lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/src/rdydata.c:240:82: warning: argument 3 of type 'PetscReal[size]' {aka 'double[size]'} declared with mismatched bound 'size' [-Wvla-parameter]
  240 | PetscErrorCode RDyGetLocalCellYCentroids(RDy rdy, const PetscInt size, PetscReal values[size]) {
      |                                                                        ~~~~~~~~~~^~~~~~~~~~~~
In file included from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/include/private/rdymmsconfigimpl.h:6,
                 from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/include/private/rdyconfigimpl.h:8,
                 from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/include/private/rdycoreimpl.h:6,
                 from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/src/rdydata.c:1:
/lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/build-frontier-gpu-opt-32bit-gcc-11-2-0-fc288817/include/rdycore.h:84:95: note: previously declared as 'PetscReal[size]' {aka 'double[size]'} with bound 'size'
   84 | PETSC_EXTERN PetscErrorCode RDyGetLocalCellYCentroids(RDy rdy, const PetscInt size, PetscReal values[size]);
      |                                                                                     ~~~~~~~~~~^~~~~~~~~~~~
/lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/src/rdydata.c:247:82: warning: argument 3 of type 'PetscReal[size]' {aka 'double[size]'} declared with mismatched bound 'size' [-Wvla-parameter]
  247 | PetscErrorCode RDyGetLocalCellZCentroids(RDy rdy, const PetscInt size, PetscReal values[size]) {
      |                                                                        ~~~~~~~~~~^~~~~~~~~~~~
In file included from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/include/private/rdymmsconfigimpl.h:6,
                 from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/include/private/rdyconfigimpl.h:8,
                 from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/include/private/rdycoreimpl.h:6,
                 from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/src/rdydata.c:1:
/lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/build-frontier-gpu-opt-32bit-gcc-11-2-0-fc288817/include/rdycore.h:85:95: note: previously declared as 'PetscReal[size]' {aka 'double[size]'} with bound 'size'
   85 | PETSC_EXTERN PetscErrorCode RDyGetLocalCellZCentroids(RDy rdy, const PetscInt size, PetscReal values[size]);
      |                                                                                     ~~~~~~~~~~^~~~~~~~~~~~
/lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/src/rdydata.c:254:77: warning: argument 3 of type 'PetscReal[size]' {aka 'double[size]'} declared with mismatched bound 'size' [-Wvla-parameter]
  254 | PetscErrorCode RDyGetLocalCellAreas(RDy rdy, const PetscInt size, PetscReal values[size]) {
      |                                                                   ~~~~~~~~~~^~~~~~~~~~~~
In file included from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/include/private/rdymmsconfigimpl.h:6,
                 from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/include/private/rdyconfigimpl.h:8,
                 from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/include/private/rdycoreimpl.h:6,
                 from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/src/rdydata.c:1:
/lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/build-frontier-gpu-opt-32bit-gcc-11-2-0-fc288817/include/rdycore.h:86:90: note: previously declared as 'PetscReal[size]' {aka 'double[size]'} with bound 'size'
   86 | PETSC_EXTERN PetscErrorCode RDyGetLocalCellAreas(RDy rdy, const PetscInt size, PetscReal values[size]);
      |                                                                                ~~~~~~~~~~^~~~~~~~~~~~
/lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/src/rdydata.c:269:81: warning: argument 3 of type 'PetscInt[size]' {aka 'int[size]'} declared with mismatched bound 'size' [-Wvla-parameter]
  269 | PetscErrorCode RDyGetLocalCellNaturalIDs(RDy rdy, const PetscInt size, PetscInt values[size]) {
      |                                                                        ~~~~~~~~~^~~~~~~~~~~~
In file included from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/include/private/rdymmsconfigimpl.h:6,
                 from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/include/private/rdyconfigimpl.h:8,
                 from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/include/private/rdycoreimpl.h:6,
                 from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/src/rdydata.c:1:
/lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/build-frontier-gpu-opt-32bit-gcc-11-2-0-fc288817/include/rdycore.h:88:94: note: previously declared as 'PetscInt[size]' {aka 'int[size]'} with bound 'size'
   88 | PETSC_EXTERN PetscErrorCode RDyGetLocalCellNaturalIDs(RDy rdy, const PetscInt size, PetscInt values[size]);
      |                                                                                     ~~~~~~~~~^~~~~~~~~~~~
/lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/src/rdydata.c:311:116: warning: argument 4 of type 'PetscReal[size]' {aka 'double[size]'} declared with mismatched bound 'size' [-Wvla-parameter]
  311 | PetscErrorCode RDyGetBoundaryEdgeXCentroids(RDy rdy, const PetscInt boundary_index, const PetscInt size, PetscReal values[size]) {
      |                                                                                                          ~~~~~~~~~~^~~~~~~~~~~~
In file included from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/include/private/rdymmsconfigimpl.h:6,
                 from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/include/private/rdyconfigimpl.h:8,
                 from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/include/private/rdycoreimpl.h:6,
                 from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/src/rdydata.c:1:
/lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/build-frontier-gpu-opt-32bit-gcc-11-2-0-fc288817/include/rdycore.h:90:129: note: previously declared as 'PetscReal[size]' {aka 'double[size]'} with bound 'size'
   90 | PETSC_EXTERN PetscErrorCode RDyGetBoundaryEdgeXCentroids(RDy rdy, const PetscInt boundary_index, const PetscInt size, PetscReal values[size]);
      |                                                                                                                       ~~~~~~~~~~^~~~~~~~~~~~
/lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/src/rdydata.c:319:116: warning: argument 4 of type 'PetscReal[size]' {aka 'double[size]'} declared with mismatched bound 'size' [-Wvla-parameter]
  319 | PetscErrorCode RDyGetBoundaryEdgeYCentroids(RDy rdy, const PetscInt boundary_index, const PetscInt size, PetscReal values[size]) {
      |                                                                                                          ~~~~~~~~~~^~~~~~~~~~~~
In file included from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/include/private/rdymmsconfigimpl.h:6,
                 from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/include/private/rdyconfigimpl.h:8,
                 from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/include/private/rdycoreimpl.h:6,
                 from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/src/rdydata.c:1:
/lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/build-frontier-gpu-opt-32bit-gcc-11-2-0-fc288817/include/rdycore.h:91:129: note: previously declared as 'PetscReal[size]' {aka 'double[size]'} with bound 'size'
   91 | PETSC_EXTERN PetscErrorCode RDyGetBoundaryEdgeYCentroids(RDy rdy, const PetscInt boundary_index, const PetscInt size, PetscReal values[size]);
      |                                                                                                                       ~~~~~~~~~~^~~~~~~~~~~~
/lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/src/rdydata.c:327:116: warning: argument 4 of type 'PetscReal[size]' {aka 'double[size]'} declared with mismatched bound 'size' [-Wvla-parameter]
  327 | PetscErrorCode RDyGetBoundaryEdgeZCentroids(RDy rdy, const PetscInt boundary_index, const PetscInt size, PetscReal values[size]) {
      |                                                                                                          ~~~~~~~~~~^~~~~~~~~~~~
In file included from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/include/private/rdymmsconfigimpl.h:6,
                 from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/include/private/rdyconfigimpl.h:8,
                 from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/include/private/rdycoreimpl.h:6,
                 from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/src/rdydata.c:1:
/lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/build-frontier-gpu-opt-32bit-gcc-11-2-0-fc288817/include/rdycore.h:92:129: note: previously declared as 'PetscReal[size]' {aka 'double[size]'} with bound 'size'
   92 | PETSC_EXTERN PetscErrorCode RDyGetBoundaryEdgeZCentroids(RDy rdy, const PetscInt boundary_index, const PetscInt size, PetscReal values[size]);
      |                                                                                                                       ~~~~~~~~~~^~~~~~~~~~~~
/lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/src/rdydata.c:335:116: warning: argument 4 of type 'PetscReal[size]' {aka 'double[size]'} declared with mismatched bound 'size' [-Wvla-parameter]
  335 | PetscErrorCode RDyGetBoundaryCellXCentroids(RDy rdy, const PetscInt boundary_index, const PetscInt size, PetscReal values[size]) {
      |                                                                                                          ~~~~~~~~~~^~~~~~~~~~~~
In file included from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/include/private/rdymmsconfigimpl.h:6,
                 from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/include/private/rdyconfigimpl.h:8,
                 from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/include/private/rdycoreimpl.h:6,
                 from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/src/rdydata.c:1:
/lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/build-frontier-gpu-opt-32bit-gcc-11-2-0-fc288817/include/rdycore.h:93:129: note: previously declared as 'PetscReal[size]' {aka 'double[size]'} with bound 'size'
   93 | PETSC_EXTERN PetscErrorCode RDyGetBoundaryCellXCentroids(RDy rdy, const PetscInt boundary_index, const PetscInt size, PetscReal values[size]);
      |                                                                                                                       ~~~~~~~~~~^~~~~~~~~~~~
/lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/src/rdydata.c:343:116: warning: argument 4 of type 'PetscReal[size]' {aka 'double[size]'} declared with mismatched bound 'size' [-Wvla-parameter]
  343 | PetscErrorCode RDyGetBoundaryCellYCentroids(RDy rdy, const PetscInt boundary_index, const PetscInt size, PetscReal values[size]) {
      |                                                                                                          ~~~~~~~~~~^~~~~~~~~~~~
In file included from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/include/private/rdymmsconfigimpl.h:6,
                 from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/include/private/rdyconfigimpl.h:8,
                 from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/include/private/rdycoreimpl.h:6,
                 from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/src/rdydata.c:1:
/lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/build-frontier-gpu-opt-32bit-gcc-11-2-0-fc288817/include/rdycore.h:94:129: note: previously declared as 'PetscReal[size]' {aka 'double[size]'} with bound 'size'
   94 | PETSC_EXTERN PetscErrorCode RDyGetBoundaryCellYCentroids(RDy rdy, const PetscInt boundary_index, const PetscInt size, PetscReal values[size]);
      |                                                                                                                       ~~~~~~~~~~^~~~~~~~~~~~
/lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/src/rdydata.c:351:116: warning: argument 4 of type 'PetscReal[size]' {aka 'double[size]'} declared with mismatched bound 'size' [-Wvla-parameter]
  351 | PetscErrorCode RDyGetBoundaryCellZCentroids(RDy rdy, const PetscInt boundary_index, const PetscInt size, PetscReal values[size]) {
      |                                                                                                          ~~~~~~~~~~^~~~~~~~~~~~
In file included from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/include/private/rdymmsconfigimpl.h:6,
                 from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/include/private/rdyconfigimpl.h:8,
                 from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/include/private/rdycoreimpl.h:6,
                 from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/src/rdydata.c:1:
/lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/build-frontier-gpu-opt-32bit-gcc-11-2-0-fc288817/include/rdycore.h:95:129: note: previously declared as 'PetscReal[size]' {aka 'double[size]'} with bound 'size'
   95 | PETSC_EXTERN PetscErrorCode RDyGetBoundaryCellZCentroids(RDy rdy, const PetscInt boundary_index, const PetscInt size, PetscReal values[size]);
      |                                                                                                                       ~~~~~~~~~~^~~~~~~~~~~~
/lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/src/rdydata.c:359:115: warning: argument 4 of type 'PetscInt[size]' {aka 'int[size]'} declared with mismatched bound 'size' [-Wvla-parameter]
  359 | PetscErrorCode RDyGetBoundaryCellNaturalIDs(RDy rdy, const PetscInt boundary_index, const PetscInt size, PetscInt values[size]) {
      |                                                                                                          ~~~~~~~~~^~~~~~~~~~~~
In file included from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/include/private/rdymmsconfigimpl.h:6,
                 from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/include/private/rdyconfigimpl.h:8,
                 from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/include/private/rdycoreimpl.h:6,
                 from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/src/rdydata.c:1:
/lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/build-frontier-gpu-opt-32bit-gcc-11-2-0-fc288817/include/rdycore.h:97:128: note: previously declared as 'PetscInt[size]' {aka 'int[size]'} with bound 'size'
   97 | PETSC_EXTERN PetscErrorCode RDyGetBoundaryCellNaturalIDs(RDy rdy, const PetscInt boundary_index, const PetscInt size, PetscInt values[size]);
      |                                                                                                                       ~~~~~~~~~^~~~~~~~~~~~
/lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/src/rdydata.c:376:82: warning: argument 3 of type 'PetscReal[size]' {aka 'double[size]'} declared with mismatched bound 'size' [-Wvla-parameter]
  376 | PetscErrorCode RDyGetLocalCellManningsNs(RDy rdy, const PetscInt size, PetscReal n_values[size]) {
      |                                                                        ~~~~~~~~~~^~~~~~~~~~~~~~
In file included from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/include/private/rdymmsconfigimpl.h:6,
                 from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/include/private/rdyconfigimpl.h:8,
                 from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/include/private/rdycoreimpl.h:6,
                 from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/src/rdydata.c:1:
/lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/build-frontier-gpu-opt-32bit-gcc-11-2-0-fc288817/include/rdycore.h:87:95: note: previously declared as 'PetscReal[size]' {aka 'double[size]'} with bound 'size'
   87 | PETSC_EXTERN PetscErrorCode RDyGetLocalCellManningsNs(RDy rdy, const PetscInt size, PetscReal values[size]);
      |                                                                                     ~~~~~~~~~~^~~~~~~~~~~~
/lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/src/rdydata.c:392:85: warning: argument 3 of type 'PetscReal[size]' {aka 'double[size]'} declared with mismatched bound 'size' [-Wvla-parameter]
  392 | PetscErrorCode RDySetManningsNForLocalCells(RDy rdy, const PetscInt size, PetscReal n_values[size]) {
      |                                                                           ~~~~~~~~~~^~~~~~~~~~~~~~
In file included from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/include/private/rdymmsconfigimpl.h:6,
                 from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/include/private/rdyconfigimpl.h:8,
                 from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/include/private/rdycoreimpl.h:6,
                 from /lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/src/rdydata.c:1:
/lustre/orion/cli192/proj-shared/gb9/rdycore/rdycore/build-frontier-gpu-opt-32bit-gcc-11-2-0-fc288817/include/rdycore.h:103:98: note: previously declared as 'PetscReal[size]' {aka 'double[size]'} with bound 'size'
  103 | PETSC_EXTERN PetscErrorCode RDySetManningsNForLocalCells(RDy rdy, const PetscInt size, PetscReal values[size]);
      |                                                                                        ~~~~~~~~~~^~~~~~~~~~~~
```

**Comment by bishtgautam** [comment]
I tried to poke around in the code, but couldn't figure out how to fix them.

**Comment by jeff-cohere** [comment]
These warnings are bogus--the declared bounds are invariably the same as the defined ones, and the issue is that the compiler seems not to be able to figure out bounds that use basic arithmetic expressions. This looks like [a known bug in early GCC 11 compilers](https://gcc.gnu.org/bugzilla/show_bug.cgi?id=97548) that looks to have been fixed in GCC 11.3. Do we have access to GCC 11.3 on these machines?

We could silence these warnings with `-Wno-vla-parameter`, but I hesitate to mask actual potential problems with VLAs that we could identify on our own workstations.

**Comment by bishtgautam** [comment]
Since these are bogus warnings, let's not add `-Wno-vla-parameter`.
